### PR TITLE
Run scheduled action for both master and release-latest

### PR DESCRIPTION
**Pull request by dabonnie**
Signed-off-by: Devin Bonnie <dbbonnie@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by dabonnie** [comment]
Current approach will not work due to https://docs.github.com/en/actions/reference/events-that-trigger-workflows#schedule

**Comment by mm318** [comment]
> Current approach will not work due to https://docs.github.com/en/actions/reference/events-that-trigger-workflows#schedule

I have an example of working around this [here](https://github.com/aws-robotics/aws-robomaker-sample-application-cloudwatch/blob/miaofei/ros1-gh-actions-enhancements/.github/workflows/main.yml#L63) (where `ros2` is the non-default branch). To keep it more elegant, we can probably even move it to a separate workflow YAML.

Or we can just skip running CI on the `release-latest` branch, since there was always mention of deprecating the `release-latest` branch when we were discussing deprecation of the `release-v*` branches.

**Comment by dabonnie** [comment]
> > Current approach will not work due to https://docs.github.com/en/actions/reference/events-that-trigger-workflows#schedule
> 
> I have an example of working around this [here](https://github.com/aws-robotics/aws-robomaker-sample-application-cloudwatch/blob/miaofei/ros1-gh-actions-enhancements/.github/workflows/main.yml#L63) (where `ros2` is the non-default branch). To keep it more elegant, we can probably even move it to a separate workflow YAML.
> 
> Or we can just skip running CI on the `release-latest` branch, since there was always mention of deprecating the `release-latest` branch when we were discussing deprecation of the `release-v*` branches.

That's exactly what I have in progress given this [thread](https://github.community/t/scheduled-builds-of-non-default-branch/16306) in a separate workflow. 

Let's deprecate when we deprecate the release branches to ensure working code.

File: .github/workflows/build_and_test_release_latest.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          ubuntu_distro: bionic
+        - ros_distro: dashing
+          ubuntu_distro: bionic
+    container:
+      image: rostooling/setup-ros-docker:ubuntu-${{ matrix.ubuntu_distro }}-ros-${{ matrix.ros_distro }}-ros-base-latest
+    outputs:
+      build_result: ${{ steps.build.outcome }}
+      test_result: ${{ steps.test.outcome }}
+    steps:
+    - uses: actions/checkout@v2
```
Comment (mm318) [inline .github/workflows/build_and_test_release_latest.yml:28]: `ros-tooling/action-ros-ci` will completely disregard what you check out with `actions/checkout` here.

Comment (dabonnie) [reply .github/workflows/build_and_test_release_latest.yml:28]: Right, shoot.

**Pull request closed without merging**